contracts/SimpleInterest.sol

```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.15;
+
+library SimpleInterest {
+    function convert(uint256 _amount, uint256 _exchangeRate, bool _inverted) public pure returns (uint256) {
+        return _inverted ? _amount * _exchangeRate : _amount / _exchangeRate;
```

Comment:
For the MVP we will not have a live data feed because there isn't a live tracker of the TST price -- since we haven't launched yet. So part of the IBCO (liquidity bootstrapping event), we are creating the on-ramp to get TST out to our community.

We cannot rely on human input to have the latest price so in the future we will surely not us this rate functionality nor a `setUnitPrice` function.

Come a time when the market prices 1 TST = 0.46 SEUR, there would be enough liquidity and trading volume. Thus we would not have to calculate the TST price like this at all but use Chainlink or similar. We could right now change to a 3 or 4 decimal system (similar to ETH with 18 dec) but the calculations would slightly increase and consequently the gas costs.

Reply:
Right now: `1/22 = 0.04545` and `1/21 = 0.04761`. It will never be exact, just as with the bonding and staking yield. Would need to increase the decimals :)

Reply:
but even for the mvp we have to track the price of the eur / usd live to price the other asset, and therefore have to do some decimal calculations there anyway. it seems like this could be an opportunity to move _all_ of our conversion calculations to one library if we just switched to using some sort of decimal system.

not only is 1/22 or 1/21 not exact, it is about 1/10th of the price!

but if we're certain that TST is definitely going to be 0.05 sEURO on launch, and we're happy calculating other asset / TST using another method, then i guess this will work for now

just feels a bit like planned obsolescence to me

Reply:
> but if we're certain that TST is definitely going to be 0.05 sEURO on launch, and we're happy calculating other asset / TST using another method, then i guess this will work for now

We are deciding the 'fair price' with the copper launch as a guideline to build up enough liquidity. This can be higher or lower than what it was last traded for.
As soon as there is enough trading activity going on and we have reliable and precise data feeds we will use them.

> just feels a bit like planned obsolescence to me

Sometimes code serves a very specific and temporal purpose. We will certainly revisit our way to calculate TST/SEUR rate down the road. You can think of this as a bit larger tick spacing in Uniswap than what you would prefer. Happy to get you on this task when we revisit it.

Reply:
i have also just noticed that the IBCO figures doc now says that 1TST = 0.055 sEURO ... so actually we need to make that default price `18, true` if we want to continue with this simple calculation, which is `0.0555... recurring`, which i guess is accurate enough but rounds up a bit ! we might want to double check with philip what impact that will have on the bonding event calculations they have?